package.json

```diff
@@ -13,13 +13,12 @@
     "./typescript": "./configs/typescript.js"
   },
   "engines": {
-    "node": ">= 18.0.0"
+    "node": ">= 20.0.0"
```

Comment:
In npm and pnpm, this config is just a warning unless setting `engine-strict=true` in `.npmrc`. But in yarn, this may occurs an error when users don't use sufficient Node.js version.

I could find only unofficial ref: [package.jsonに"engines"を設定すると「このバージョンのNode.jsでしか動かない」を表明できる #JavaScript - Qiita](https://qiita.com/suin/items/994458418c737cc9c3e8), let me check actual behavior of yarn before you merge it.

Reply:
- npm: https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines
- pnpm: https://pnpm.io/package_json#engines
- yarn v1: https://classic.yarnpkg.com/en/docs/package-json#engines-
- yarn 2+: I couldn't find the doc about `engines` behavior.


npm and pnpm say:
> Unless the user has set the [engine-strict config](https://docs.npmjs.com/cli/v10/using-npm/config#engine-strict) flag, this field is advisory only and will only produce warnings when your package is installed as a dependency.

However, yarn doesn't say so.

Reply:
In the local project with Node 16 & yarn v1, I tried to install next@latest only supported over Node 18, and then the error came up:

```
❯ yarn add next@latest
yarn add v1.22.19
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
error next@14.2.2: The engine "node" is incompatible with this module. Expected version ">=18.17.0". Got "16.20.0"
error Found incompatible module.
```

**Therefore, also this PR change will force yarn users to use Node.js over 20 when they install the package to their project.**

Reply:
@wakamsha Please do one of the following approaches:
1. Revert `engines` field config to use Node 18 & Downgrade semantic-release to v22.x to be supported Node 18
2. Remove `engines` field

Reply:
Ah, exactly. I will update the approaches.